service/handler.go

```diff
@@ -179,25 +180,36 @@ func (h *HandlerService) Handle(conn net.Conn) {
 	}()
 
 	// read loop
-	buf := make([]byte, 2048)
+	data := make([]byte, constants.IOBufferBytesSize)
+	buf := bytes.NewBuffer(nil)
 	for {
-		n, err := conn.Read(buf)
-		if err != nil {
-			logger.Log.Debugf("Read message error: '%s', session will be closed immediately", err.Error())
-			return
+		totalLen := 0
+		n := len(data)
+		var err error
+		for n == len(data) {
+			n, err = conn.Read(data)
```

Comment:
What happens if Read returns one and a half packets? I think currently it would throw the second packet away. Shouldn't we shift the remaining bytes to the beginning of the buffer and append with the next read call? (I don't know if that happens in practice, in theory I think it should)

Reply:
I guess this issue somewhat relates to what @recklessMo said